### PR TITLE
Refactors packet code

### PR DESCRIPTION
**Pull request by AngellusMortis**
* Removes all exceptions to new module `pyunifiprotect.exceptions`
* Adds new base classes for exceptions
* Cleans up imports from `pyunifiprotect` to be relative
* Adds new class enum `ModelType` to for `modelKey` values
* Adds refactors decoding of packets from Unifi Protect into new classes: `WSPacketFrameHeader`, `WSPacketFrame`, and `WSPacket`
    * These classes will give a more structured approach to handling packets from Protect
    * Basic support for updating/creating new Protect packets
* Adds a test to make sure `decode_ws_frame` stays consistent / working

### CODE REVIEW

File: pyunifiprotect/unifi_data.py [hunk 1]
```diff
@@ -777,3 +776,176 @@ def __setitem__(self, key, value):
...
+        # b: unknown
+        # i: payload_size
+
+        header_end = position + WS_HEADER_SIZE
+
+        try:
+            packet_type, payload_format, deflated, unknown, payload_size = struct.unpack(
+                "!bbbbi", data[position:header_end]
+            )
+        except Exception as e:
```
Comment (bdraco) [inline pyunifiprotect/unifi_data.py:847]: Since this is only wrapping `struct.unpack` can you narrow the exception handler to `struct.error`?

Comment (AngellusMortis) [reply pyunifiprotect/unifi_data.py:847]: Probably a great idea. I just wanted to make sure anything that happened was re-raised as a predictable error.

**Review by bdraco — COMMENTED** [review]
This looks good. I'll test this as soon.

**Review by bdraco — APPROVED** [review]
Testing looks good 👍

**Comment by bdraco** [comment]
The tests passes, I did have to install the new version as it didn't work in-place.  Thats probably not an issue right now since the CI isn't running the tests anyways.
```
# pytest
Test session starts (platform: linux, Python 3.8.11, pytest 6.2.4, pytest-sugar 0.9.4)
rootdir: /root/pyunifiprotect, configfile: pyproject.toml, testpaths: tests
plugins: aiohttp-0.3.0, profiling-1.7.0, parallel-0.1.0, anyio-1.3.0, forked-1.1.3, mock-3.1.1, test-groups-1.0.3, requests-mock-1.8.0, reportlog-0.1.2, rerunfailures-9.1.1, sugar-0.9.4, timeout-1.4.2, xdist-2.1.0, repeat-0.9.1, raises-0.11, asyncio-0.15.1, aresponses-2.1.4, socket-0.4.0, cov-2.12.0
collecting ... 
 tests/test_unifi_protect_server.py ✓✓                                                                                                                                                                                       100% ██████████

Results (0.08s):
       2 passed
```

**Comment by AngellusMortis** [comment]
CI not running tests makes complete sense. That is why it did not break when I forgot `pillow` on the other branch. Do you want me to fix them real fast?

**Comment by bdraco** [comment]
That would be great.  Please do it in a fresh PR.

I'll merge this now.

**Comment by bdraco** [comment]
Screen Shot 2021-08-29 at 7 45 14 PM

Web socket still works

**Pull request merged**